src/graphql/controller.ts

```diff
@@ -205,34 +219,79 @@ export class GqlEntityController {
       const tableName = pluralize(type.name.toLowerCase());
 
       builder = builder.dropTableIfExists(tableName).createTable(tableName, t => {
-        t.primary(['id']);
+        //if there is no autoIncrement fields on current table, mark the id as primary
+        if (
+          autoIncrementFieldsMap.size == 0 ||
+          autoIncrementFieldsMap.get(tableName)?.length === 0
+        ) {
+          t.primary(['id']);
+        }
 
         this.getTypeFields(type).forEach(field => {
           const fieldType = field.type instanceof GraphQLNonNull ? field.type.ofType : field.type;
           if (isListType(fieldType) && fieldType.ofType instanceof GraphQLObjectType) return;
-          const sqlType = this.getSqlType(field.type);
-
-          let column =
-            'options' in sqlType
-              ? t[sqlType.name](field.name, ...sqlType.options)
-              : t[sqlType.name](field.name);
-
-          if (field.type instanceof GraphQLNonNull) {
-            column = column.notNullable();
-          }
-
-          if (!['text', 'json'].includes(sqlType.name)) {
-            column.index();
+          //Check if field is declared as autoincrement
+          if (autoIncrementFieldsMap.get(tableName)?.includes(field.name)) {
+            t.increments(field.name, { primaryKey: true });
+          } else {
+            const sqlType = this.getSqlType(field.type);
+            let column =
+              'options' in sqlType
+                ? t[sqlType.name](field.name, ...sqlType.options)
+                : t[sqlType.name](field.name);
+
+            if (field.type instanceof GraphQLNonNull) {
+              column = column.notNullable();
+            }
+
+            if (!['text', 'json'].includes(sqlType.name)) {
+              column.index();
+            }
           }
         });
       });
     });
 
     await builder;
-
     return { builder };
   }
 
+  /**
+   * Parse schema to extract table and fields witrh annotation autoIncrement
+   * @param schema
+   * @returns A map where key is table name and values a list of fields with annotation autoIncrement
+   */
+  private extractAutoIncrementFields(schema: string) {
```

Comment:
This shouldn't be handled with regexes. Directive can be handled natively with GraphQL.

Reply:
As we extended GraphQL with our custom directive [here](https://github.com/checkpoint-labs/checkpoint/blob/23698a4587b4fe2b1160fe0224401e0d5d1bb90f/src/utils/graphql.ts#L12-L16), we can then access that directive directly from our GraphQL tree.

This way you can look at the field (id field in this example, and check if it has `@autoincrement` directive, same as we do with `@derivedFrom`:
https://github.com/checkpoint-labs/checkpoint/blob/e6dd8b3580604d39de79f4c8c41fb2103582677c/src/graphql/resolvers.ts#L182-L186

That means we don't have to bother with writing our own dummy parser for GraphQL to extract autoIncrementFields, and just check for `field.astNode.directives` here:
https://github.com/checkpoint-labs/checkpoint/blob/23698a4587b4fe2b1160fe0224401e0d5d1bb90f/src/graphql/controller.ts#L223-L228

Reply:
OK thx for the hint. I will check it and modify accordingly.